Importando as bibliotecas e módulos necessários: 

In [27]:
#!pip install pandas
#!pip install openpyxl
import os
import pandas as pd
import numpy as np
import sys
import gc

Os datasets analisados neste notebook são as Bases de dados de Alunos: Os microdados de alunos do SAEB são apresentados em base de dados agrupadas segundo a série avaliada. Dessa forma, as bases de dados dos alunos são apresentadas separadamente, incluindo dados de respostas e resultados dos testes de Língua Portuguesa, de Matemática, de Ciências Humanas e de Ciências da Natureza, bem como as respostas aos questionários. Cada uma das bases traz o conjunto completo de alunos participantes naquela série. Analisaremos apenas os dados dos alunos do 9º ano do Ensino Fundamental.

### 1. Coleta de dados 
Esta seção descreve o processo de coleta de dados realizado para construir os conjuntos de dados.

### 1.1 Estrutura do conjunto de dados
Esta seção descreve as estruturas e conteúdos dos conjuntos de dados utilizados, incluindo o número de linhas e colunas e descrições das observações
e variáveis. Eles foram carregados em um pandas.DataFrame, e como já estão fornecidos como um arquivo CSV, podem ser carregados facilmente usando a 
função read_csv().

#### SAEB 2019
O dataset de 2019 será tomado como base para os demais anos desse estudo.

In [5]:
gc.collect
df_saeb_2019 = pd.read_csv("2019/TS_ALUNO_9EF.csv",  encoding='latin-1', sep=";", dtype=str)
df_saeb_2019

,ID_SAEB,ID_REGIAO,ID_UF,ID_MUNICIPIO,ID_AREA,ID_ESCOLA,IN_PUBLICA,ID_LOCALIZACAO,ID_TURMA,ID_SERIE,...,TX_RESP_Q016,TX_RESP_Q017A,TX_RESP_Q017B,TX_RESP_Q017C,TX_RESP_Q017D,TX_RESP_Q017E,TX_RESP_Q018A,TX_RESP_Q018B,TX_RESP_Q018C,TX_RESP_Q019
0,2019,1,11,6311030,2,61255632,1,1,1030927,9,...,A,D,A,D,B,A,C,B,A,C
1,2019,1,11,6311030,2,61255632,1,1,1030927,9,...,A,D,A,D,B,A,A,A,A,C
2,2019,1,11,6311030,2,61255632,1,1,1030927,9,...,A,D,A,D,C,A,C,B,B,D
3,2019,1,11,6311030,2,61255632,1,1,1030927,9,...,A,C,A,D,C,A,A,A,A,D
4,2019,1,11,6311030,2,61255632,1,1,1030927,9,...,A,C,A,*,B,A,B,A,A,C
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2388926,2019,5,53,6316599,1,61322272,1,1,1067040,9,...,.,.,.,.,.,.,.,.,.,.
2388927,2019,5,53,6316599,1,61322272,1,1,1067040,9,...,A,D,B,C,D,A,B,C,B,A
2388928,2019,5,53,6316599,1,61322272,1,1,1067040,9,...,A,D,A,A,C,A,B,C,B,C
2388929,2019,5,53,6316599,1,61322272,1,1,1067040,9,...,A,D,A,B,B,A,B,.,.,B


Informações gerais sobre o conjunto de dados:
Como ele contém mais de 100 colunas, o parâmetro verbose de info() foi definido como True para exibir todas as colunas de dados. O parâmetro show_counts também foi explicitamente definido como True para mostrar a contagem de valores não nulos por coluna.

In [26]:
df_saeb_2019.info(verbose=True, show_counts=True)

<class 'pandas.DataFrame'>
RangeIndex: 2388931 entries, 0 to 2388930
Data columns (total 129 columns):
 #    Column                         Non-Null Count    Dtype
---   ------                         --------------    -----
 0    ID_SAEB                        2388931 non-null  str  
 1    ID_REGIAO                      2388931 non-null  str  
 2    ID_UF                          2388931 non-null  str  
 3    ID_MUNICIPIO                   2388931 non-null  str  
 4    ID_AREA                        2388931 non-null  str  
 5    ID_ESCOLA                      2388931 non-null  str  
 6    IN_PUBLICA                     2388931 non-null  str  
 7    ID_LOCALIZACAO                 2388931 non-null  str  
 8    ID_TURMA                       2388931 non-null  str  
 9    ID_SERIE                       2388931 non-null  str  
 10   ID_ALUNO                       2388931 non-null  str  
 11   IN_SITUACAO_CENSO              2388931 non-null  str  
 12   IN_PREENCHIMENTO_LP            238893

O conjunto de dados contém 2388931 observações (linhas) em 129 variáveis (colunas). Cada observação corresponde a um aluno do 9° ano participante na 
avaliação em 2019, e as variáveis correspondem a coleta de informações que possam avaliar os eixos do SAEB como equidade (nível socioeconômico dos estudantes, perfil individual e familiar) e atendimento escolar (acesso e trajetória escolar) e desempenho nas provas.
As variáveis e esquemas de codificação para respostas categóricas do dataset do ano de 2019 são os seguintes:

In [31]:
dicionario_2019 = pd.read_excel(r"2019/Dicionario_Saeb_2019.xlsx", sheet_name="TS_ALUNO_9EF")
dicionario_2019.columns = ["Variável", "Tipo", "Tamanho", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta"]
dicionario_2019 = dicionario_2019[["Variável", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta" ]]

#dicionario_2019 = dicionario_2019[["TS_ALUNO_9EF", "Unnamed: 3", "Unnamed: 4", "Unnamed: 5", "Unnamed: 6" ]]
#dicionario_2019 = dicionario_2019.dropna(how="all")

from IPython.display import display
display(dicionario_2019.style.hide(axis="index"))

Variável,Descrição,Código de Preenchimento,Código da opção de resposta,Texto da opção de resposta
nan,nan,nan,nan,nan
Variável,Descrição,Código de Preenchimento,nan,nan
ID_SAEB,Ano de aplicação do Saeb,2019,nan,nan
ID_REGIAO,Código da Região,1 - Norte,nan,nan
nan,nan,2 - Nordeste,nan,nan
nan,nan,3 - Sudeste,nan,nan
nan,nan,4 - Sul,nan,nan
nan,nan,5 - Centro-Oeste,nan,nan
ID_UF,Código da Unidade da Federação,11-RO,nan,nan
nan,nan,12-AC,nan,nan


<!-- Abaixo está uma listagem literal das perguntas do Questionário de Estudante(TX_RESP_Q001 a TX_RESP_Q019). Os esquemas de codificação para as respostas categóricas também estão enumerados:
#### ID_SAEB
- 2019
#### ID_REGIAO	
- 1: Norte
- 2: Nordeste
- 3: Sudeste
- 4: Sul
- 5: Centro-Oeste
#### ID_UF	
- 11: RO
- 12: AC
- 13: AM
- 14: RR
- 15: PA
- 16: AP
- 17: TO
- 21: MA
- 22: PI
- 23: CE
- 24: RN
- 25: PB
- 26: PE
- 27: AL
- 28: SE
- 29: BA
- 31: MG
- 32: ES
- 33: RJ
- 35: SP
- 41: PR
- 42: SC 
- 43: RS
- 50: MS
- 51: MT
- 52: GO
- 53: DF
#### ID_AREA	
- 1: Capital
- 2: Interior
#### IN_PUBLICA	
- 0: Não pública
- 1: Pública
#### ID_LOCALIZACAO	
- 1: Urbana
- 2: Rural
#### ID_SERIE	
- 9: 9° ano do Ensino Fundamental
#### IN_SITUACAO_CENSO
- 0: Não consistente
- 1: Consistente
#### IN_PREENCHIMENTO_LP	
- 0: Prova não prenchida
- 1: Prova preenchida
#### IN_PREENCHIMENTO_MT
- 0: Prova não prenchida
- 1: Prova preenchida
#### IN_PREENCHIMENTO_CH	
- 0: Prova não prenchida
- 1: Prova preenchida
- Vazio: Não selecionado para a amostra de CH
#### IN_PREENCHIMENTO_CN	
- 0: Prova não prenchida
- 1: Prova preenchida
- Vazio: Não selecionado para a amostra de CN
#### IN_PRESENCA_LP	
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_MT
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_CH
- 0: Ausente
- 1: Presente
#### IN_PRESENCA_CN
- 0: Ausente
- 1: Presente
#### ID_CADERNO_LP
- Prova Regular (Cadernos 1 a 21)
- Macrotipo 18 (Caderno 22)
#### ID_BLOCO_1_LP	
- De 1 a 7 (8 no macrotipo 18 - caderno 22)
#### ID_BLOCO_2_LP	
- De 1 a 7 (9 no macrotipo 18 - caderno 22)
#### ID_CADERNO_MT
- Prova Regular (Cadernos 1 a 21)
- Macrotipo 18 (Caderno 22)
#### ID_BLOCO_1_MT	
- De 1 a 7 (8 no macrotipo 18 - caderno 22)
#### ID_BLOCO_2_MT
- De 1 a 7 (9 no macrotipo 18 - caderno 22)
#### ID_CADERNO_CH
- Prova Regular (Cadernos 1 a 26)
- Macrotipo 18 (Caderno 27)
#### ID_BLOCO_1_CH	
- De 1 a 13
#### ID_BLOCO_2_CH	
- De 1 a 13
#### ID_BLOCO_3_CH	
- De 1 a 13
#### NU_BLOCO_1_ABERTA_CH	
- De 1 a 6
#### NU_BLOCO_2_ABERTA_CH
- De 1 a 6
#### ID_CADERNO_CN	
- Prova Regular (Cadernos 1 a 14)
- Macrotipo 18 (Caderno 15)
#### ID_BLOCO_1_CN	
- De 1 a 7
#### ID_BLOCO_2_CN	
- De 1 a 7
#### ID_BLOCO_3_CN	
- De 1 a 7
#### NU_BLOCO_1_ABERTA_CN	
- De 1 a 3
#### NU_BLOCO_2_ABERTA_CN
- De 1 a 3
#### TX_RESP_BLOCO_1_LP	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_LP	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_1_MT	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_MT	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_1_CH
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_2_CH	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO_3_CH	
- A, B, C, D, . (branco), * (nulo)
#### CO_CONCEITO_Q1_CH	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### CO_CONCEITO_Q2_CH	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### TX_RESP_BLOCO1_CN	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO2_CN	
- A, B, C, D, . (branco), * (nulo)
#### TX_RESP_BLOCO3_CN	
- A, B, C, D, . (branco), * (nulo)
#### CO_CONCEITO_Q1_CN	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### CO_CONCEITO_Q2_CN	
- 0: Nenhum crédito
- 1: Crédito parcial
- 2: Crédito total
- 7: Erros de impressão ou digitalização
- .: Branco
#### IN_PROFICIENCIA_LP	
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_MT	
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_CH
- 0: Não
- 1: Sim
#### IN_PROFICIENCIA_CN	
- 0: Não
- 1: Sim
#### IN_AMOSTRA
- 0: Não
- 1: Sim
#### ESTRATO	
- Os estratos são compostos por características da participação da escola na avaliação e representam agrupamentos para os quais a avaliação fornece resultados confiáveis para Língua Portuguesa e Matemática. 
#### ESTRATO_CIENCIAS
- Os estratos são compostos por características da participação da escola na avaliação e representam agrupamentos para os quais a avaliação fornece resultados confiáveis para Ciências Humanas e Ciências da Natureza. 
#### PESO_ALUNO_LP
- Valor com 7 casas decimais 
#### PROFICIENCIA_LP	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_LP
- Valor com 7 casas decimais 
#### PROFICIENCIA_LP_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_LP_SAEB	
- Valor com 7 casas decimais 
#### PESO_ALUNO_MT
- Valor com 7 casas decimais 
#### PROFICIENCIA_MT	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_MT	
- Valor com 7 casas decimais 
#### PROFICIENCIA_MT_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_MT_SAEB
- Valor com 7 casas decimais 
#### PESO_ALUNO_CH	
- Valor com 7 casas decimais 
#### PROFICIENCIA_CH
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CH	
- Valor com 7 casas decimais 
#### PROFICIENCIA_CH_SAEB	
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CH_SAEB	
- Valor com 7 casas decimais 
#### PESO_ALUNO_CN
- Valor com 7 casas decimais 
#### PROFICIENCIA_CN
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CN
- Valor com 7 casas decimais 
#### PROFICIENCIA_CN_SAEB
- Valor com 7 casas decimais 
#### ERRO_PADRAO_CN_SAEB
- Valor com 7 casas decimais
#### IN_PREENCHIMENTO_QUESTIONARIO
- 0: Não preenchido
- 1: Preenchido parcial ou totalmente
#### TX_RESP_Q001	- Qual língua você fala com mais frequência em sua casa?
- A: Português
- B: Espanhol
- C: Outra língua
#### TX_RESP_Q002 - Qual é a sua cor ou raça?
- A: Branca
- B: Preta
- C: Parda
- D: Amarela
- E: Indígena
- F: Não quero declarar
#### TX_RESP_Q003a - Normalmente, quem mora na sua casa? - Mãe (mães ou madrasta)
- A: Não
- B: Sim
#### TX_RESP_Q003b - Normalmente, quem mora na sua casa? - Pai (pais ou padrasto)
- A: Não
- B: Sim
#### TX_RESP_Q003c -  Normalmente, quem mora na sua casa? - Irmão(s) ou irmã(s)
- A: Não
- B: Sim
#### TX_RESP_Q003d - Normalmente, quem mora na sua casa? - Avô ou avó
- A: Não
- B: Sim
#### TX_RESP_Q003e -  Normalmente, quem mora na sua casa? - Outros (tios, primos, etc.)
- A: Não
- B: Sim
#### TX_RESP_Q004 - Qual é a maior escolaridade da sua mãe (ou mulher responsável por você)?
- A: Não completou o 5° ano do Ensino Fundamental
- B: Ensino Fundamental, até o 5° ano
- C: Ensino Fundamental completo
- D: Ensino Médio completo
- E: Ensino Superior completo (faculdade ou graduação)
- F: Não sei
#### TX_RESP_Q005 - Qual é a maior escolaridade do seu pai (ou homem responsável por você)?
- A: Não completou o 5° ano do Ensino Fundamental
- B: Ensino Fundamental, até o 5° ano
- C: Ensino Fundamental completo
- D: Ensino Médio completo
- E: Ensino Superior completo (faculdade ou graduação)
- F: Não sei
#### TX_RESP_Q006a - Com que frequência seus pais ou responsáveis costumam: - Conversar com você sobre o que acontece na escola
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q006b -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a estudar
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q006c -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a fazer a tarefa de casa
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q006d -  Com que frequência seus pais ou responsáveis costumam: - Incentivar você a comparecer às aulas
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q006e -  Com que frequência seus pais ou responsáveis costumam: - Ir às reuniões de pais na escola
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q007 - Com que frequência sua família paga alguém para auxiliar nos trabalhos domésticos (faxina ou limpeza)?
- A: Nunca ou quase nunca
- B: De vez em quando (uma vez por semana, a cada quinze dias, etc.)
- C: Sempre ou quase sempre (ex.:três ou mais dias por semana)
#### TX_RESP_Q008a - Na região que você mora tem: Rua pavimentada (asfalto ou calçamento)
- A: Não
- B: Sim
#### TX_RESP_Q008b - Na região que você mora tem: Água tratada da rua
- A: Não
- B: Sim
#### TX_RESP_Q008c - Na rua que você mora tem: Iluminação na rua
- A: Não
- B: Sim
#### TX_RESP_Q009a - Dos itens relacionados abaixo, quantos existem na sua casa? - Geladeira
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009b - Dos itens relacionados abaixo, quantos existem na sua casa? - Tablet
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009c - Dos itens relacionados abaixo, quantos existem na sua casa? - Computador (ou notebook)
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009d - Dos itens relacionados abaixo, quantos existem na sua casa? - Quartos para dormir
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009e - Dos itens relacionados abaixo, quantos existem na sua casa? - Televisão
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009f - Dos itens relacionados abaixo, quantos existem na sua casa? - Banheiro
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q009g - Dos itens relacionados abaixo, quantos existem na sua casa? - Carro
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q010a - Na sua casa tem: - Tv a cabo (Ex.:Net, Sky, etc.)
- A: Não
- B: Sim
#### TX_RESP_Q010b - Na sua casa tem: - Rede Wi-Fi
- A: Não
- B: Sim
#### TX_RESP_Q010c - Na sua casa tem: - Um quarto só seu
- A: Não
- B: Sim
#### TX_RESP_Q010d - Na sua casa tem: - Mesa para estudar (ou escrivaninha)
- A: Não
- B: Sim
#### TX_RESP_Q010e - Na sua casa tem: - Garagem
- A: Não
- B: Sim
#### TX_RESP_Q010f - Na sua casa tem: - Forno de microondas
- A: Não
- B: Sim
#### TX_RESP_Q010g - Na sua casa tem: - Aspirador de pó
- A: Não
- B: Sim
#### TX_RESP_Q010h - Na sua casa tem: - Máquina de lavar roupa 
- A: Não
- B: Sim
#### TX_RESP_Q010i - Na sua casa tem: - Freezer (independente ou segunda porta da geladeira)
- A: Não
- B: Sim
#### TX_RESP_Q011 - Quanto tempo você demora para chegar à sua escola?
- A: Menos de 30 minutos
- B: Entre 30 minutos e uma hora
- C: Mais de uma hora
#### TX_RESP_Q012 - Considerando a maior distãncia percorrida, normalmente de que forma você chega à sua escola?
- A: À pé
- B: De De ônibus urbano 
- C: De transporte escolar
- D: De barco
- E: De bicicleta
- F: De carro
- G: Outros meios de transporte
#### TX_RESP_Q013 - Com que idade você entrou na escola?
- A: 3 anos ou menos
- B: 4 ou 5 anos 
- C: 6 ou 7 anos
- D: 8 anos ou mais
#### TX_RESP_Q014 - A partir do primeiro ano do ensino fundamental, em que tipo de escola você estudou?
- A: Somente em escola pública
- B: Somente em escola particular
- C: Em escola pública e em escola particular
#### TX_RESP_Q015 - Você já foi reprovado(a)?
- A: Não
- B: Sim, uma vez
- C: Sim, duas vezes ou mais
#### TX_RESP_Q016 - Alguma vez você abandonou a escola deixando de frequentá-la até o final do ano escolar?
- A: Nunca
- B: Sim, uma vez
- C: Sim, duas vezes ou mais
#### TX_RESP_Q017a - Fora da escola em dias de aula, quanto tempo você usa para: - Lazer (TV, internet, jogar bola, música, etc.)
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q017b - Fora da escola em dias de aula, quanto tempo você usa para: - Fazer cursos
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q017c - Fora da escola em dias de aula, quanto tempo você usa para: - Fazer trabalhos domésticos (lavar louça, limpar quintal, cuidar dos irmãos)
- *: Nulo
- .: Branco
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q017d - Fora da escola em dias de aula, quanto tempo você usa para: - Estudar (lição de casa, trabalhos escolares, etc.)
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q017e - Fora da escola em dias de aula, quanto tempo você usa para: - Trabalhar fora de casa (recebendo ou não um salário)
- A: Não uso meu tempo pra isso
- B: Menos de 1 hora
- C: Entre 1 e 2 horas
- D: Mais de 2 horas
#### TX_RESP_Q018a - Com que frequência você costuma: - Ler notícias (jornais, revistas, internet, etc.) 
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q018b - Com que frequência você costuma: - Ler livros que não sejam das matérias escolares
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q018c - Com que frequência você costuma: - Ler histórias em quadrinhos (mangás, gibis, etc.)
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q019 - Quando terminar o Ensino Fundamental você pretende
- A: Somente continuar estudando
- B: Somente trabalhar
- C: Continuar estudando e trabalhar
- D: Ainda não sei
-->

### 1.2 Obtenção dos dados dos demais anos

#### SAEB 2021

Algumas variáveis e questões se diferenciam em relação ao ano de 2019, que é o ano base, ou seja, o dataset de 2021 apresenta variáveis e questões que foram incluídas na avaliação desse respectivo ano. Nesse caso, para o dataset de 2021 será apresentado apenas aquilo que foi acrescentado em relação ao ano de 2019:  
- Variáveis:
    - IN_INSE (Indicador para cálculo do INSE (São considerados válidos os estudantes que responderam pelo menos 8 itens, dentre os 17 utilizados para o cálculo do indicador))
    - INSE_ALUNO (Resultado individual do INSE para o aluno)
    - NU_TIPO_NIVEL_INSE (Classificação do Indicador de Nível Socioeconômico em 8 Grupos)
    - PESO_ALUNO_INSE (Peso do Aluno para cálculo do INSE 2021)		 	 
- Questões:
    - TX_RESP_Q01 (Qual é o seu sexo?)
    - TX_RESP_Q02 (Qual é a sua idade?)
    - TX_RESP_Q05 (Você possui algum tipo de necessidade especial?)
    - TX_RESP_Q09a (Com que frequência seus pais ou responsáveis costumam: - Ler em casa.)
    - TX_RESP_Q11h (Dos itens relacionados abaixo, quantos existem na sua casa? - Celular com internet (smartphone).)
    - TX_RESP_Q15 (Você se utiliza de transporte escolar, ou passe escolar, para ir à escola?)
    - TX_RESP_Q22a (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu possuía equipamento adequado para acompanhar o ensino remoto.)
    - TX_RESP_Q22b (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu tive conexão de internet adequada para acesso às aulas remotas.)
    - TX_RESP_Q22c (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Tive facilidade em usar os programas de comunicação nas aulas remotas.)
    - TX_RESP_Q22d (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Recebi material impresso da escola (livros, apostilas, atividades em folha etc.)	
    - TX_RESP_Q22e (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Os professores me auxiliaram a entender o conteúdo.)
    - TX_RESP_Q22f (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu compreendi o conteúdo das aulas remotas.)
    - TX_RESP_Q22g (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Em casa havia um lugar tranquilo para eu assistir às aulas.)
    - TX_RESP_Q22h (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus familiares apoiaram o meu estudo.)
    - TX_RESP_Q22i (Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus colegas me apoiaram durante o ensino remoto.)

As variáveis e esquemas de codificação para respostas categóricas do dataset do ano de 2021 são os seguintes:


In [32]:
dicionario_2021 = pd.read_excel(r"2021/Dicionario_Saeb_2021.xlsx", sheet_name="TS_ALUNO_9EF")

dicionario_2021.columns = ["Variável", "Tipo", "Tamanho", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta"]
dicionario_2021 = dicionario_2021[["Variável", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta" ]]


#dicionario_2021 = dicionario_2021[["TS_ALUNO_9EF", "Unnamed: 3", "Unnamed: 4", "Unnamed: 5", "Unnamed: 6" ]]

linhas_selecionadas = [136,137,138,139,140,141,142,143,144,145,146,147,149,150,151,152,153,161,162,163,164,165,166,167,168,190,191,192,193,230,231,232,233,
                       234,314,315,316,317,318,319,370,371,372,373,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,
                       452,453,454,455,456,457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483,
                       484]
dicionario_filt = dicionario_2021.iloc[linhas_selecionadas]

from IPython.display import display
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)
display(dicionario_filt)


,Variável,Descrição,Código de Preenchimento,Código da opção de resposta,Texto da opção de resposta
136,IN_INSE,"Indicador para cálculo do INSE (São considerados válidos os estudantes que responderam pelo menos 8 itens, dentre os 17 utilizados para o cálculo do indicador)",0 - Não,NaN,NaN
137,NaN,NaN,1 - Sim,NaN,NaN
138,INSE_ALUNO,Resultado individual do INSE para o aluno,NaN,NaN,NaN
139,NU_TIPO_NIVEL_INSE,"Classificação do Indicador de Nível Socioeconômico em 8 Grupos (para melhor entendimento dos grupos, consultar a nota técnica disponível no portal do Inep em <https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/indicadores-educacionais/nivel-socioeconomico>)",1 - Nível I,NaN,NaN
140,NaN,NaN,2 - Nível II,NaN,NaN
141,NaN,NaN,3 - Nível III,NaN,NaN
142,NaN,NaN,4 - Nível IV,NaN,NaN
143,NaN,NaN,5 - Nível V,NaN,NaN
144,NaN,NaN,6 - Nível VI,NaN,NaN
145,NaN,NaN,7 - Nível VII,NaN,NaN


<!-- Abaixo estão mostrados os esquemas de codificação para as respostas categóricas e uma listagem literal das perguntas do Questionário de Estudante, apenas aqueles que se diferenciam em relação ao ano de 2019:  
#### ID_SAEB
- 2021
#### IN_INSE - Indicador para cálculo do INSE (São considerados válidos os estudantes que responderam pelo menos 8 itens, dentre os 17 utilizados para o cálculo do indicador)
- 0: Não
- 1: Sim
#### NU_TIPO_NIVEL_INSE - Classificação do Indicador de Nível Socioeconômico em 8 Grupos 
- 1: Nível I
- 2: Nível II
- 3: Nível III
- 4: Nível IV
- 5: Nível V
- 6: Nível VI
- 7: Nível VII
- 8: Nível VIII
#### PESO_ALUNO_INSE - Peso do Aluno para cálculo do INSE 2021
- Valor com 7 casas decimais
#### TX_RESP_Q01 - Qual é o seu sexo?
- *: Nulo
- .: Branco
- A: Masculino
- B: Feminino
#### TX_RESP_Q02 - Qual é a sua idade?
- *: Nulo
- .: Branco
- A: 13 anos ou menos
- B: 14 anos
- C: 15 anos
- D: 16 anos
- E: 17 anos
- F: 18 anos ou mais
#### TX_RESP_Q05 - Você possui algum tipo de necessidade especial?
- *: Nulo
- .: Branco
- A: Sim
- B: Não
#### TX_RESP_Q09a - Com que frequência seus pais ou responsáveis costumam: - Ler em casa
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q11h - Dos itens relacionados abaixo, quantos existem na sua casa? - Celular com internet (smartphone)
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q15 - Você se utiliza de transporte escolar, ou passe escolar, para ir à escola?
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q22a - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu possuía equipamento adequado para acompanhar o ensino remoto
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22b - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu tive conexão de internet adequada para acesso às aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22c - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Tive facilidade em usar os programas de comunicação nas aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22d - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Recebi material impresso da escola (livros, apostilas, atividades em folha, etc.) 
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22e - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Os professores me auxiliaram a entender o conteúdo
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22f - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Eu compreendi o conteúdo das aulas remotas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22g - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Em casa havia um lugar tranquilo para eu assistir às aulas
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22h - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus familiares apoiaram o meu estudo
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
#### TX_RESP_Q22i - Durante a pandemia, indique a frequência com que os seguintes fatos ocorreram: - Meus colegas me apoiaram durante o ensino remoto
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Várias vezes
- D: Sempre
-->

#### SAEB 2023

Algumas variáveis e questões se diferenciam em relação ao ano de 2019, que é o ano base, ou seja, o dataset de 2023 apresenta variáveis e questões que foram incluídas na avaliação desse respectivo ano. Nesse caso, para o dataset de 2023 será apresentado apenas aquilo que foi acrescentado em relação ao ano de 2019:  
- Variáveis:
    - IN_INSE (Indicador para cálculo do INSE (São considerados válidos os estudantes que responderam pelo menos 8 itens, dentre os 17 utilizados para o cálculo do indicador))
    - INSE_ALUNO (Resultado individual do INSE para o aluno)
    - NU_TIPO_NIVEL_INSE (Classificação do Indicador de Nível Socioeconômico em 8 Grupos)
    - PESO_ALUNO_INSE (Peso do Aluno para cálculo do INSE 2021) 	 
- Questões:
    - TX_RESP_Q01 (Qual é o seu sexo?)
    - TX_RESP_Q02 (Qual é a sua idade?)
    - TX_RESP_Q05a (Você possui deficiência, transtorno do espectro autista ou superdotação? - Deficiência.)
    - TX_RESP_Q05b (Você possui deficiência, transtorno do espectro autista ou superdotação? - Transtorno do espectro autista.)
    - TX_RESP_Q05c (Você possui deficiência, transtorno do espectro autista ou superdotação? - Altas habilidades ou superdotação.)
    - TX_RESP_Q06 (Quantas pessoas moram na sua casa, contando com você?)
    - TX_RESP_Q10a (Com que frequência seus pais ou responsáveis costumam: - Ler em casa.)
    - TX_RESP_Q12g (Dos itens relacionados abaixo, quantos existem na sua casa? - Celular com internet (smartphone).)
    - TX_RESP_Q15a (Você utiliza para ir à escola: - Transporte gratuito escolar.)
    - TX_RESP_Q15b (Você utiliza para ir à escola: - Passe escolar.)
    - TX_RESP_Q22a (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - No início do ano, eles(as) informaram sobre o que seria ensinado e aprendido?)
    - TX_RESP_Q22b (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Antes de iniciar um novo conteúdo, eles(as) perguntam o que vocês sabem sobre o conteúdo?)
    - TX_RESP_Q22c (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) trazem temas do cotidiano para serem debatidos em sala de aula?)
    - TX_RESP_Q22d (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas sobre desigualdade racial?)
    - TX_RESP_Q22e (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas sobre desigualdade de gênero?)
    - TX_RESP_Q22f (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas como bullying e outras formas de violência?)
    - TX_RESP_Q22g (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) desenvolvem trabalhos em grupos?)
    - TX_RESP_Q22h (Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam questões relacionadas ao futuro profissional dos(as) estudantes?)
    - TX_RESP_Q23a (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me interesso sobre o que foi ensinado na escola neste ano.)
    - TX_RESP_Q23b (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto motivado(a), no dia a dia, a usar o que foi ensinado.)
    - TX_RESP_Q23c (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Há espaço para diferentes opiniões na minha sala de aula.)
    - TX_RESP_Q23d (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto seguro(a) quando estou na escola.)
    - TX_RESP_Q23e (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto à vontade para discordar dos(as) meus(minhas) professores(as).)
    - TX_RESP_Q23f (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu consigo argumentar sobre conteúdos difíceis.)
    - TX_RESP_Q23g (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Os resultados das avaliações representam o quanto eu aprendi.)
    - TX_RESP_Q23h (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: -Meus (Minhas) professores(as) acreditam que eu sou capaz de aprender.)
    - TX_RESP_Q23i (Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: -Meus (Minhas) professores(as) me motivam a continuar meus estudos.) 

As variáveis e esquemas de codificação para respostas categóricas do dataset do ano de 2023 são os seguintes:


In [40]:
dicionario_2023 = pd.read_excel(r"2023/Dicionario_Saeb_2023.xlsx", sheet_name="TS_ALUNO_9EF")

dicionario_2023.columns = ["Variável", "Tipo", "Tamanho", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta"]
dicionario_2023 = dicionario_2023[["Variável", "Descrição", "Código de Preenchimento", "Código da opção de resposta", "Texto da opção de resposta" ]]

#dicionario_2023 = dicionario_2023[["TS_ALUNO_9EF", "Unnamed: 3", "Unnamed: 4", "Unnamed: 5", "Unnamed: 6" ]]

linhas_selecionadas = [134,135,136,137,138,139,140,141,142,143,144,145,147,148,149,150,151,152,153,154,155,156,157,158,159,174,175,176,177,178,179,180,
                       181,182,183,184,185,186,187,188,189,190,191,192,229,230,231,232,233,307,308,309,310,311,312,354,355,356,357,358,359,360,361,424,
                       425,426,427,428,429,430,431,432,433,434,435,436,437,438,439,440,441,442,443,444,445,446,447,448,449,450,451,452,453,454,455,456,
                       457,458,459,460,461,462,463,464,465,466,467,468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483,484,485,486,487,488,
                       489,490,491,492,493,494,495,496,497,498,499,500,501,502,503,504,505,506,507,508,509,510,511,512,513,514,515,516,517,518,519,520,
                       521,522,523,524,525]
dicionario_filt = dicionario_2023.iloc[linhas_selecionadas]

from IPython.display import display
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)
display(dicionario_filt)

,Variável,Descrição,Código de Preenchimento,Código da opção de resposta,Texto da opção de resposta
134,IN_INSE,"Indicador para cálculo do INSE (São considerados válidos os estudantes que responderam pelo menos 8 itens, dentre os 17 utilizados para o cálculo do indicador)",0 - Não,NaN,NaN
135,NaN,NaN,1 - Sim,NaN,NaN
136,INSE_ALUNO,Resultado individual do INSE para o aluno,NaN,NaN,NaN
137,NU_TIPO_NIVEL_INSE,"Classificação do Indicador de Nível Socioeconômico em 8 Grupos (para melhor entendimento dos grupos, consultar a nota técnica disponível no portal do Inep em <https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/indicadores-educacionais/nivel-socioeconomico>)",1 - Nível I,NaN,NaN
138,NaN,NaN,2 - Nível II,NaN,NaN
139,NaN,NaN,3 - Nível III,NaN,NaN
140,NaN,NaN,4 - Nível IV,NaN,NaN
141,NaN,NaN,5 - Nível V,NaN,NaN
142,NaN,NaN,6 - Nível VI,NaN,NaN
143,NaN,NaN,7 - Nível VII,NaN,NaN


<!-- Abaixo estão mostrados os esquemas de codificação para as respostas categóricas e uma listagem literal das perguntas do Questionário de Estudante, apenas aqueles que se diferenciam em relação ao ano de 2019:  
#### ID_SAEB
- 2023
#### IN_INSE	
- 0: Não
- 1: Sim
#### NU_TIPO_NIVEL_INSE	
- 1: Nível I
- 2: Nível II
- 3: Nível III
- 4: Nível IV
- 5: Nível V
- 6: Nível VI
- 7: Nível VII
- 8: Nível VIII
#### PESO_ALUNO_INSE
- Valor com 7 casas decimais
#### TX_RESP_Q01 - Qual é o seu sexo?
- *: Nulo
- .: Branco
- A: Masculino
- B: Feminino
- C: Não quero declarar
#### TX_RESP_Q02 - Qual é a sua idade?
- *: Nulo
- .: Branco
- A: 13 anos ou menos
- B: 14 anos
- C: 15 anos
- D: 16 anos
- E: 17 anos
- F: 18 anos ou mais
#### TX_RESP_Q10a - Com que frequência seus pais ou responsáveis costumam: - Ler em casa
- *: Nulo
- .: Branco
- A: Nunca ou quase nunca
- B: De vez em quando
- C: Sempre ou quase sempre
#### TX_RESP_Q12g - Dos itens relacionados abaixo, quantos existem na sua casa? - Celular com internet (smartphone)
- *: Nulo
- .: Branco
- A: Nenhum
- B: 1
- C: 2
- D: 3 ou mais
#### TX_RESP_Q15a - Você utiliza para ir à escola: - Transporte gratuito escolar
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q15b - Você utiliza para ir à escola: - Passe escolar
- *: Nulo
- .: Branco
- A: Não
- B: Sim
#### TX_RESP_Q22a - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - No início do ano, eles(as) informaram sobre o que seria ensinado e aprendido?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22b - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Antes de iniciar um novo conteúdo, eles(as) perguntam o que vocês sabem sobre o conteúdo?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22c - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) trazem temas do cotidiano para serem debatidos em sala de aula?
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22d - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas sobre desigualdade racial?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22e - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas sobre desigualdade de gênero?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22f - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam temas como bullying e outras formas de violência?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22g - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) desenvolvem trabalhos em grupos?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q22h - Para os próximos itens, indique qual é a proporção de professores(as) da sua turma que abordam os seguintes temas em sala de aula: - Eles(as) abordam questões relacionadas ao futuro profissional dos(as) estudantes?
- *: Nulo
- .: Branco
- A: Todos eles
- B: A maior parte deles
- C: Poucos deles
- D: Nenhum deles
#### TX_RESP_Q23a - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me interesso sobre o que foi ensinado na escola neste ano
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23b - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto motivado(a), no dia a dia, a usar o que foi ensinado
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23c - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Há espaço para diferentes opiniões na minha sala de aula
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23d - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto seguro(a) quando estou na escola
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23e - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu me sinto à vontade para discordar dos(das) meus(minhas) professores(as)
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23f - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Eu consigo argumentar sobre conteúdos difíceis
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23g - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Os resultados das avaliações representam o quanto eu aprendi
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23h - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Meus(Minhas) professores(as) acreditam que eu sou capaz de aprender
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
#### TX_RESP_Q23i - Sobre sua escola, indique o quanto você concorda ou discorda das afirmações abaixo: - Meus(Minhas) professores(as) me motivam a continuar meus estudos
- *: Nulo
- .: Branco
- A: Concordo totalmente 
- B: Concordo
- C: Discordo
- D: Discordo totalmente
-->